addons/web/static/src/scss/bootstrap_review.scss

```diff
@@ -195,3 +205,67 @@ $o-navbar-nav-pills-link-border-radius: $nav-pills-border-radius !default;
 .carousel-control-next .visually-hidden {
     left: 50%; // Avoid horizontal scrollbar in Chrome
 }
+
+// Modal
+.modal-content {
+    // If the text color of the body (used for the text color in modals) is not
+    // visible due to insufficient contrast with the modal background, we adjust
+    // the text color in the modal using the following code. For example, if the
+    // user sets a black background for its website and the text color of the
+    // body is white, the text will not be visible on modals with a white
+    // background.
+
+    // Check if the body text color is light.
+    $-is-body-text-light: color-contrast($body-color) == $color-contrast-dark;
+
+    @if $modal-content-color == null {
+        color: if(
+            $-is-body-text-light,
+            color-contrast($modal-content-bg, $body-color, $color-contrast-dark),
+            color-contrast($modal-content-bg, $color-contrast-light, $body-color)
+        );
+
+        .table-striped > tbody > tr:nth-of-type(odd) > * {
+            // When the website background is dark (resulting in light-colored
+            // body text), the text color of the odd rows of striped tables
+            // matches the body text color. Consequently, it becomes invisible
+            // on the light background of a modal. To address this, we inherit
+            // the text color from 'modal-content,' which is also tied to the
+            // body text color but adapts appropriately within modals (see
+            // above).
+            color: inherit;
+        }
+        // This prevents these elements from taking their colors from the body
+        // inside a modal.
+        @if $input-bg == $body-bg {
+            .form-control {
+                background-color: transparent;
+            }
+        }
+        @if $input-color == $body-color {
+            .form-control {
+                color: inherit;
+            }
+        }
+        @if $form-select-bg == $body-bg {
+            .form-select {
+                background-color: transparent;
+            }
+        }
+        @if $form-select-color == $body-color {
+            .form-select {
+                color: inherit;
+            }
+        }
+        @if $form-check-input-bg == $body-bg {
+            .form-check-input:not(:checked) {
+                background-color: transparent;
+            }
+        }
```

Comment:
All these lines make this change:

BEFORE
![Capture d’écran du 2024-01-22 09-45-22](https://github.com/odoo/odoo/assets/52911687/92a6547c-7d17-49dd-9d73-498b3991d7b5)

AFTER
![Capture d’écran du 2024-01-22 09-44-59](https://github.com/odoo/odoo/assets/52911687/55fd2cf4-b29b-4f39-b9a6-2b11d3cec088)